src/components/Calculator.js

```diff
@@ -1,49 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/Calculate';
 
-function CalBtn(props) {
-  const { incrementNumber, bgColor } = props;
-  return (<button type="button" className={bgColor}>{incrementNumber}</button>);
-}
+function Calculator() {
+  const [calcData, setCalcData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-CalBtn.propTypes = {
-  incrementNumber: PropTypes.number.isRequired,
-  bgColor: PropTypes.string.isRequired,
-};
+  function callActions(e) {
+    const newValueInput = calculate(calcData, e.target.value);
+    setCalcData(newValueInput);
+  }
+  function CalBtn(props) {
+    const {
+      incrementNumber, characterStr, bgColor,
+    } = props;
+
+    return (<input type="button" onClick={callActions} className={bgColor} value={incrementNumber || characterStr} />);
+  }
+
+  const initialValue = calcData.next || calcData.total || 0;
+  CalBtn.propTypes = {
+    incrementNumber: PropTypes.number,
+    characterStr: PropTypes.string,
+    bgColor: PropTypes.string.isRequired,
+  };
+
+  CalBtn.defaultProps = {
+    incrementNumber: 0,
+    characterStr: '',
+  };
 
-function Calculator() {
   return (
     <section className="container">
-      <input disabled value={0} className="show-input" />
+      <input type="text" className="show-input" value={initialValue} readOnly />
       <aside>
         <div>
-          <CalBtn incrementNumber="AC" bgColor="bg-white btn p-normal" />
-          <CalBtn incrementNumber="+/-" bgColor="bg-white btn p-normal" />
-          <CalBtn incrementNumber="%" bgColor="bg-white btn p-normal" />
-          <CalBtn incrementNumber="+" bgColor="bg-orange btn p-normal" />
+          <CalBtn characterStr="AC" bgColor="bg-white btn p-normal" />
+          <CalBtn characterStr="+/-" bgColor="bg-white btn p-normal" />
+          <CalBtn characterStr="%" bgColor="bg-white btn p-normal" />
+          <CalBtn characterStr="+" bgColor="bg-orange btn p-normal" />
```

Comment:
- [ ] I saw there are two `+` signs in your calculator but there should be one `+` and division sign `÷ or /` was missing :fearful: Kindly remove one plus and add one division sign in your app to make your calculator do all the events :+1: